src/decoder/mod.rs

```diff
@@ -804,9 +804,12 @@ impl<R: Read + Seek> Decoder<R> {
             }
         };
 
-        if bytes / buffer.byte_len() > max_uncompressed_length {
+        if bytes / buffer.byte_len() < max_uncompressed_length {
```

Comment:
I'm not 100% sure about this yet. It seems illogical to check if we had read _more_ data but currently we avoid this in `lzw` at least.

Reply:
This reveals an interesting test failure. `u12` data is packed in bytes but the decoder uses the `buffer.byte_len()` and multiplies it with the sample count to determine the expected buffer size. That's incorrect. Instead, the strip's data representation should be independent of the BitsPerSample.

Reply:
I think that the original code for this line was correct. Specifically for reasons I don't really understand, `max_uncompressed_length` is being used as the size for `buffer`. This check is to make sure that the subsequent indexing operations on buffer do not panic.

The case of too little data being read is caught by the calling function https://github.com/image-rs/image-tiff/blob/c9703dcb8a008567c5d9c6263195f56850e253a7/src/decoder/mod.rs#L1038

Reply:
Hm, [this Stackexchange answer](https://stackoverflow.com/questions/7068183/writing-10-12-bit-tiff-files-with-libtiff-c) suggest that while libTIFF packs the bits, there is no actually standard prescribed behavior. We might have to adjust for that?

Reply:
Isn't any `u12` or similar data going to trigger UnsupportedColorType error below?

Reply:
The problem is, this triggers _first_ and if the other change is reverted than we must check the units_read.